lib/src/license_detection/primary_filter.dart

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+import 'package:pana/src/license_detection/license.dart';
+
+/// Filters [knownLicenses] and returns a list of licenses that could likely be a possible match for input.
+///
+/// A license might be a possible match from the list of [knownLicenses].
+/// This function filters from the known licenses by weighing the
+/// number of similar tokens through [tokenSimilarity] method and
+/// returns a list of [License] which have a score above certain threshold.
+List<LicenseWithNGrams> filter(
+  Map<String, int> occurrences,
+  List<License> knownLicenses,
+  int granularity,
+) {
+  var possibleLicenses = <LicenseWithNGrams>[];
+  knownLicenses
+      .where((license) =>
+          tokenSimilarity(
+            occurrences,
+            license.occurrences,
+          ) >=
+          0.5)
```

Comment:
Where does this threshold come from? Can we explain why we thought this was a good threshold? or add a link?